include/picongpu/particles/atomicPhysics/kernel/UpdateElectricField.kernel

```diff
@@ -0,0 +1,126 @@
+         * @param fieldEnergyUseCacheBox deviceDataBox giving access to the field energy use cache for each local
+         *  superCell
+         */
+        template<
+            typename T_Worker,
+            typename T_AreaMapping,
+            typename T_TimeRemainingBox,
+            typename T_EFieldBox,
+            typename T_FieldEnergyUseCacheBox>
+        HDINLINE void operator()(
+            T_Worker const& worker,
+            T_AreaMapping const areaMapping,
+            T_TimeRemainingBox const timeRemainingBox,
+            T_EFieldBox eFieldBox,
+            T_FieldEnergyUseCacheBox const fieldEnergyUseCacheBox) const
+        {
+            auto const superCellIdx = KernelIndexation::getSuperCellIndex(worker, areaMapping);
+            auto const superCellFieldIdx = KernelIndexation::getSuperCellFieldIndex(worker, areaMapping, superCellIdx);
+
+            auto const timeRemaining = timeRemainingBox(superCellFieldIdx);
+            if(timeRemaining <= 0._X)
+                return;
+
+            T_FieldEnergyUseCache const& eFieldEnergyUseCache = fieldEnergyUseCacheBox(superCellFieldIdx);
+            DataSpace<picongpu::simDim> const superCellCellOffset = superCellIdx * picongpu::SuperCellSize::toRT();
+
+            auto forEachCell = pmacc::lockstep::makeForEach<T_FieldEnergyUseCache::numberCells, T_Worker>(worker);
+            forEachCell(
+                [&worker, &superCellCellOffset, &eFieldBox, &eFieldEnergyUseCache](uint32_t const linearCellIndex)
+                {
+                    DataSpace<picongpu::simDim> const localCellIndex
+                        = pmacc::math::mapToND(picongpu::SuperCellSize::toRT(), static_cast<int>(linearCellIndex));
+                    DataSpace<picongpu::simDim> const cellIndex = localCellIndex + superCellCellOffset;
+
+                    // ((sim.unit.mass() * sim.unit.length())/(sim.unit.time()^2 * sim.unit.charge()))^2
+                    auto const oldEFieldVector = eFieldBox(cellIndex);
+                    auto const eFieldNormSquared = pmacc::math::l2norm2(oldEFieldVector);
+
+                    /// @note zero field also means at most zero energy use
+                    if(eFieldNormSquared <= 0)
+                        return;
+
+                    // eV * 1 = eV * sim.unit.energy()/sim.unit.energy() = (ev / sim.unit.energy()) * sim.unit.energy()
+                    // sim.unit.energy()
+                    float_X const eFieldEnergyUse
+                        = picongpu::sim.pic.get_eV() * eFieldEnergyUseCache.energyUsed(linearCellIndex);
+
+                    // sim.unit.charge()^2 * sim.unit.time()^2 / (sim.unit.mass() * sim.unit.length()^3)
+                    //  * sim.unit.length()^3
+                    // = sim.unit.charge()^2 * sim.unit.time()^2 / sim.unit.mass()
+                    constexpr float_X eps0HalfTimesCellVolume
+                        = (picongpu::sim.pic.getEps0() / 2._X) * picongpu::sim.pic.getCellSize().productOfComponents();
```

Comment:
(1) You performed this calculation already somewhere else.
Why is it repeated here?

(2) In the other calculation you directly used `eps0HalfTimesCellVolume`.